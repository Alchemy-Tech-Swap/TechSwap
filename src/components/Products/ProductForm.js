import React from 'react';

export default function ProductForm({
  title,
  description,
  price,
  image,
  catagories,
  condition,
  onChange,
  onSubmit,
}) {
  return (
    <form onSubmit={onSubmit}>
      <label>Title:</label>
      <input id="title" name="title" type="text" value={title} onChange={onChange} />

      <label>Description:</label>
      <input
        id="description"
        name="description"
        type="text"
        value={description}
        onChange={onChange}
      />

      <label>Price:</label>
      <input id="price" name="price" type="text" value={price} onChange={onChange} />

      <label>Image:</label>
      <input id="image" name="image" type="text" value={image} onChange={onChange} />

      <label>Catagory:</label>
      <input id="catagories" name="catagories" type="text" value={catagories} onChange={onChange} />

      <label>Condition:</label>
      <input id="condition" name="condition" type="text" value={condition} onChange={onChange} />
    </form>
  );
}
