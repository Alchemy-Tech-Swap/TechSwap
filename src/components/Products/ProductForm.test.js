import ProductForm from './ProductForm';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom/cjs/react-router-dom.min';

test('Should render Product Form', async () => {
  const { container } = render(
    <MemoryRouter>
      <ProductForm
        products={{
          onChange: 'ƒ onChange() {}',
          onSubmit: 'ƒ onSubmit() {}',
        }}
      />
    </MemoryRouter>
  );
  await screen.findByText('Condition:');
  expect(container).toMatchSnapshot();
});
