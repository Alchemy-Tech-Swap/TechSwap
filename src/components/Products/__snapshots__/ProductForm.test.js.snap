// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Product Form 1`] = `
<div>
  <form>
    <label>
      Title:
    </label>
    <input
      id="title"
      name="title"
      type="text"
      value=""
    />
    <label>
      Description:
    </label>
    <input
      id="description"
      name="description"
      type="text"
      value=""
    />
    <label>
      Price:
    </label>
    <input
      id="price"
      name="price"
      type="number"
      value=""
    />
    <label>
      Image:
    </label>
    <input
      id="image"
      name="file"
      type="file"
    />
    <label>
      Category:
    </label>
    <input
      id="categories"
      name="categories"
      type="text"
      value=""
    />
    <label>
      Condition:
    </label>
    <input
      id="condition"
      name="condition"
      type="text"
      value=""
    />
    <button>
      Submit
    </button>
  </form>
</div>
`;
